src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10664,30 +10664,94 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
-                    leftOperand = this.AddError(leftOperand, ErrorCode.ERR_ConditionalInInterpolation);
+                    // if we now are at a colon, this was preferred parse.  
+                    whenTrue = newWhenTrue;
                 }
                 else
                 {
-                    var colon = this.EatToken(SyntaxKind.ColonToken);
-                    var colonRight = this.ParsePossibleRefExpression();
-                    leftOperand = _syntaxFactory.ConditionalExpression(leftOperand, questionToken, colonLeft, colon, colonRight);
+                    // retrying the parse didn't help.  Use the original interpretation.
+                    originalAfterWhenTrue.Reset();
                 }
             }
 
-            return leftOperand;
+            if (this.CurrentToken.Kind == SyntaxKind.EndOfFileToken && this.lexer.InterpolationFollowedByColon)
+            {
+                // We have an interpolated string with an interpolation that contains a conditional expression.
+                // Unfortunately, the precedence demands that the colon is considered to signal the start of the
+                // format string. Without this code, the compiler would complain about a missing colon, and point
+                // to the colon that is present, which would be confusing. We aim to give a better error message.
+                leftOperand = _syntaxFactory.ConditionalExpression(
+                    leftOperand,
+                    questionToken,
+                    whenTrue,
+                    SyntaxFactory.MissingToken(SyntaxKind.ColonToken),
+                    _syntaxFactory.IdentifierName(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)));
+                return this.AddError(leftOperand, ErrorCode.ERR_ConditionalInInterpolation);
+            }
+            else
+            {
+                return _syntaxFactory.ConditionalExpression(
+                    leftOperand,
+                    questionToken,
+                    whenTrue,
+                    this.EatToken(SyntaxKind.ColonToken),
+                    this.ParsePossibleRefExpression());
+            }
+
+            static bool containsTernaryCollectionToReinterpret(ExpressionSyntax expression)
+            {
+                var stack = ArrayBuilder<GreenNode>.GetInstance();
+                stack.Push(expression);
+
+                while (stack.Count > 0)
+                {
+                    var current = stack.Pop();
+                    if (current is ConditionalExpressionSyntax conditionalExpression &&
+                        conditionalExpression.WhenTrue.GetFirstToken().Kind == SyntaxKind.OpenBracketToken)
+                    {
+                        stack.Free();
+                        return true;
+                    }
+
+                    foreach (var child in current.ChildNodesAndTokens())
```

Comment:
Do we need to not go through child lambdas/anonymous delegate expressions?

Reply:
I'm not sure if it's possible to _observe_ the effect, as we reset `ForceConditionalAccessExpression` when entering a lambda, but it seems like we'll do unnecessary reparses.

Reply:
right.  we do not need to go into those.  i could add similar logic here.  but it does make things more fragile.  Given how exceptionally rare any of these cases should be, my prefernece was simplicity versus fragile precision.

Reply:
Alright. May be worth a comment calling it out as an explicit decision to avoid fragility.

Reply:
will do.

Reply:
done.